sdk/storage/azure-storage-blob/README.md

```diff
@@ -187,6 +187,54 @@ async for blob in container.list_blobs():
 print(blob_list)
 ```
 
+## Optional Configuration
+
+Optional keyword arguments that can be passed in at the client and per-operation level. 
+
+### Retry Policy configuration
+
+Use the following keyword arguments when instantiating a client to configure the retry policy:
+
+* _retry_total_ (int): Total number of retries to allow. Takes precedence over other counts.
+    Pass in `retry_total=0` if you do not want to retry on requests. Defaults to 10.
+* _retry_connect_ (int): How many connection-related errors to retry on. Defaults to 3.
+* _retry_read_ (int): How many times to retry on read errors. Defaults to 3.
+* _retry_status_ (int): How many times to retry on bad status codes. Defaults to 3.
+* _retry_to_secondary_ (bool): Whether the request should be retried to secondary, if able.
+    This should only be enabled of RA-GRS accounts are used and potentially stale data can be handled.
+    Defaults to False.
+
+### Encryption configuration
+
+Use the following keyword arguments when instantiating a client to configure encryption:
+
+* _require_encryption_ (bool): If set to True, will enforce that objects are encrypted and decrypt them.
+* _key_encryption_key_ (object): The user-provided key-encryption-key. Must implement the following methods:
+    - `wrap_key(key)`--wraps the specified key using an algorithm of the user's choice. 
+    - `get_key_wrap_algorithm()`--returns the algorithm used to wrap the specified symmetric key.
+    - `get_kid()`--returns a string key id for this key-encryption-key.
+* _key_resolver_function_ (callable): The user-provided key resolver. Uses the kid string to return a key-encryption-key
+    implementing the interface defined above.
```

Comment:
I'm not sure if we wanted to document client-side encryption, as I think we want to discourage its use. @johanste?

Reply:
+1